### PR TITLE
Replace some GX constants with enums

### PR DESCRIPTION
**Pull request by 1011X**
This change groups together related constants and adds the following new data types:
+ `enum CmpFn`
+ `enum AlphaOp`
+ `enum Primitive`
+ `enum BlendMode`
+ `enum LogicOp`
+ `enum BlendCtrl`

It also modifies some GX functions to take `bool` instead of `GX_TRUE` and `GX_FALSE`.

### CODE REVIEW

File: src/gx.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 //! This module implements a safe wrapper around the graphics functions found in ``gx.h``.
 
 use core::ffi::c_void;
+use crate::ffi;
 use ogc_sys::{Mtx as Mtx34, Mtx44};
```
Comment (ProfElements) [inline src/gx.rs:7]: Please keep all imports off the same crate in the same use. EX instead of `use ogc_sys`, use the crate:ffi` for everything

Comment (1011X) [reply src/gx.rs]: Okay, I've replaced all uses of `ogc_sys` with `ffi`

**Comment by ProfElements** [comment]
LGTM

**Comment by FrictionlessPortals** [comment]
LGTM, thanks!

**Pull request merged**